BDD/src/step-definitions/steps.js

```diff
@@ -0,0 +1,88 @@
+import { Given, When, Then, And } from '@wdio/cucumber-framework';
+import { expect, $, browser } from '@wdio/globals';
+
+import { MainPage } from '../pages/mainPage';
+import { DashboardPage } from '../pages/dashboardPage';
+import { LoginHelper } from '../utils/helpers/loginHelper';
+
+const mainPage = new MainPage();
+const dashboardPage = new DashboardPage();
+const loginHelper = new LoginHelper();
+
+Given('I open web site {string}', async page => {
+	await mainPage.open(page);
+	await mainPage.maximize();
+});
+
+When(
+	'I login with default User credentials {string} and {string}',
+	async (username, password) => {
+		await loginHelper.login(username, password);
+	}
+);
+
+When('Dashboard icon displays on the left side bar', async () => {
+	await expect(mainPage.dashboardIcon).toBeDisplayed();
+});
+
+When(
+	'I click on Dashboard icon then Add New Dashboard button displays',
+	async () => {
+		await mainPage.dashboardIcon.isDisplayed();
+		await mainPage.dashboardIcon.click();
+		await dashboardPage.addNewDashboardBtn.isDisplayed();
+	}
+);
+
+Then(
+	'I click on Add New Dashboard button and verify popup appears',
+	async () => {
+		await dashboardPage.addNewDashboardBtn.click();
+		await dashboardPage.addDashboardModalWindow.waitForDisplayed({
+			timeout: 2000,
+		});
+		await expect(
+			await dashboardPage.addDashboardModalWindow.isDisplayed()
```

Comment:
It's not necessary to overcomplicate things here. Instead of using the long-form `await expect(await dashboardPage.addDashboardModalWindow.isDisplayed()).toEqual(true);,` you could write it more concisely as `await expect(dashboardPage.addDashboardModalWindow).toBeDisplayed().` I’d recommend applying this approach to all assertions to make the code cleaner and simpler.